api/admin/controller/individual_admin_settings.py

```diff
@@ -57,25 +54,42 @@ def process_post(self):
         else:
             roles = []
 
-        roles_error = self.handle_roles(user, admin, roles, settingUp)
+        roles_error = self.handle_roles(admin, roles, settingUp)
         if roles_error:
             return roles_error
 
         password = flask.request.form.get("password")
-        self.handle_password(password, admin, is_new, user)
+        self.handle_password(password, admin, is_new, settingUp)
 
         return self.response(admin, is_new)
 
     def check_permissions(self, admin, settingUp):
         """Before going any further, check that the user actually has permission
          to create/edit this type of admin"""
-        # User must be a sitewide library manager in order to create/edit
-        # a sitewide library manager.
-        if admin.is_sitewide_library_manager() and not settingUp:
-            self.require_sitewide_library_manager()
-        # User must be a system admin in order to create/edit a system admin.
-        if admin.is_system_admin() and not settingUp:
-            self.require_system_admin()
+
+         # For readability: the person who is submitting the form is referred to as "user"
+         # rather than as something that could be confused with "admin" (the admin
+         # which the user is submitting the form in order to create/edit.)
+
+        if not settingUp:
+            user = flask.request.admin
+
+            # System admin has all permissions.
+            if user.is_system_admin():
+                return
+
+            # If we've hit this point, then the user isn't a system admin.  If the
+            # admin is a system admin, the user won't be able to do anything.
+            if admin.is_system_admin():
+                raise AdminNotAuthorized
```

Comment:
Maybe it doesn't matter but I think there are usually parentheses after this - this is raising an exception class rather than an instance of it?

Reply:
Good catch :)